internal/bminventory/inventory.go

```diff
@@ -1535,7 +1537,7 @@ func (b *bareMetalInventory) generateClusterInstallConfig(ctx context.Context, c
 
 	installerReleaseImageOverride := ""
 	if isBaremetalBinaryFromAnotherReleaseImageRequired(cluster.CPUArchitecture, cluster.OpenshiftVersion, cluster.Platform.Type) {
-		defaultArchImage, err := b.versionsHandler.GetReleaseImage(cluster.OpenshiftVersion, common.DefaultCPUArchitecture)
+		defaultArchImage, err := b.versionsHandler.GetReleaseImage(cluster.OpenshiftVersion, cluster.CPUArchitecture)
```

Comment:
Instead of modifying the content of this conditional, could we maybe modify the content of the function `isBaremetalBinaryFromAnotherReleaseImageRequired` so that it does not enter this code at all? AFAIK this block is meant to support cross-installation where Hub is x86 but spoke is ARM (but not the reverse). I.e. this block handles a scenario when `openshift-install` is executed on platform x86 but manifests that it generate may be used by ARM. By writing here `cluster.CPUArchitecture` you are saying "use openshift-install binary that matches the installed cluster architecture" which is different from "use openshift-install binary that matches the Hub cluster architecture"

Reply:
TLDR: Have you tested that with this change you can install single-arch ARM cluster from X86 hub with no regression?